File: app/src/main/java/com/hop/pirate/callback/ResultCallBack.java
package com.hop.pirate.callback;

/**
 * @description:
 * @author: mr.x
 * @date :   2020/5/23 6:30 PM
 */
public interface ResultCallBack<T> {
   void onError(Throwable e);
    void onSuccess(T t);
    void onComplete();
}

File: app/src/main/java/com/hop/pirate/callback/ResultCallBack.kt
package com.hop.pirate.callback

/**
 * @description:
 * @author: mr.x
 * @date :   2020/5/23 6:30 PM
 */
interface ResultCallBack<T> {
    fun onError(e: Throwable?)
    fun onSuccess(t: T)
    fun onComplete()
}